runtime/bundles/org.eclipse.core.runtime/src/org/eclipse/core/runtime/ILog.java

```diff
@@ -175,4 +176,40 @@ public static ILog get() {
 			return of(ILog.class);
 		}
 	}
+
+	/**
+	 * Adds a log listener that is notified about all log events of the runtime.
+	 * <p>
+	 * Once registered, a listener starts receiving notification as entries are
+	 * added via any <code>ILog.log()</code>. The listener continues to receive
+	 * notifications until it is replaced or removed.
+	 * </p>
+	 *
+	 * @param listener the listener to register
+	 * @see ILog#addLogListener(ILogListener)
+	 * @see #removeLogListener(ILogListener)
+	 * @since 3.31
+	 */
+	public static void addRuntimeLogListener(ILogListener listener) {
+		if (listener == null) {
+			return;
+		}
+		RuntimeLog.addLogListener(listener);
```

Comment:
I don't think adding method dereferencing a static helps a lot as it doesn't really make it clearer what happens nor does is make things much simpler. The "RuntimeLog" is a particular instance of the log, internal to Platform, I don't really get why ILog should have particular method to manipulate this particular instance.
Can you please add to this PR examples of usage where this API addition helps noticeably? What about adding a `public static ILog runtimeLog()` accessor if we often need it?

Reply:
The point is to replace` org.eclipse.core.runtime.Platform.addLogListener(ILogListener)` so people are not need to be tied to the "Platform" class an not to invent / provide really new / different things.

Reply:
OK, so then having a `public static ILog runtimeLog()` would work, wouldn't it?
My concern with those 2 methods is that they start to mix the responsibility of this interface: it used to be the representation of a Log, then we added public accessors for various Logs (that's a good pattern to provide most probable accessors in the interface, so it's fine too), but this particular methods would then add the responsibility of managing 1 particular Log instance, which seems less consistent.
If we can have users doing `ILog.runtimeLog().addLogListener()` that's probably good and clear enough; without mixing duties in the interface.

Reply:
I'm not sure if we can/should expose the runtime log, the intention seem to be making it only internally accessible but allowing other code to listen for events logged there.